frontend/taipy-gui/src/components/Taipy/Notification.tsx

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import Alert from "@mui/material/Alert";  
+import { TaipyBaseProps } from "./utils";
+
+interface NotificationProps extends TaipyBaseProps {
+    severity?: "error" | "warning" | "info" | "success";
+    message: string | (() => string);  // Dynamic string handling
+    variant?: "filled" | "outlined" | "standard";
+}
+
+const Notification = (props: NotificationProps) => {  
+    const { severity = "info", message, variant = "filled" } = props;
+
+    // Handle dynamic string by checking if `message` is a function and calling it
+    const displayMessage = typeof message === "function" ? message() : message;
```

Comment:
const displayMessage = useDynamicProperty(props.message, props.defaultMessage, undefined); you could take a look into our other components to see how dynamic properties are being handled